### PR TITLE
Feature/authentication code review

### PR DESCRIPTION
**Pull request by majovainilla**
Requesting review for the project about Authentication in Rails.

This project includes:

- [x] Descriptive Readme file

- [x] Stickler files and error checking

- [x] Sessions and Sign In

- [x] Authentication and Posts

- [x] Sign Up

- [x] Production version (see link in Readme)

### CODE REVIEW

**Comment by majovainilla** [comment]
The Stickler is showing an error about an unused argument. But it's an argument needed to make the gravatar work. So we didn't delete it.

File: app/helpers/sessions_helper.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def remember(user)
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user.remember_token
+  end
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply app/helpers/sessions_helper.rb]: Also, please, don't use `user.id`

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module UsersHelper
+  def gravatar_for(user, size: 60)
+    gravatar_id = Digest::MD5.hexdigest(user.email.downcase)
+    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}"
+    image_tag(gravatar_url, alt: user.name, class: 'gravatar')
+  end
```
[the remark this answers is not in the document]
Comment (fabricio-garcia) [reply app/helpers/users_helper.rb]: - [ ] Solved

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
```
[the remark this answers is not in the document]
Comment (majovainilla) [reply app/helpers/sessions_helper.rb]: OK!

File: app/helpers/sessions_helper.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user&.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
+  end
```
[the remark this answers is not in the document]
Comment (majovainilla) [reply app/helpers/sessions_helper.rb]: ok!

File: app/helpers/sessions_helper.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (majovainilla) [reply app/helpers/sessions_helper.rb]: Done!

File: app/helpers/sessions_helper.rb [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module SessionsHelper
+  def log_in(user)
+    user.remember
+    cookies.permanent[:remember_token] = user.remember_me
+  end
+
+  def current_user
+    @current_user ||= User.find_by(remember_me: cookies[:remember_token])
```
Comment (samgaco) [inline app/helpers/sessions_helper.rb]: :+1:

**Pull request merged**